### ISSUE TITLE
feature flags treatment always keeping the initial value

### ISSUE THREAD

**Issue by michaellegare** [splitio/dotnet-client#206]
### .Net version
.net 6

### Library Version
Every version after `7.4.3`

### Issue
Every version past `7.4.3` seems to never get an updated value of the treatment. It always keeps the value that was fetched initially no matter the value of `FeaturesRefreshRate` in the configuration.

To demonstrate this I have the following console application code:
```csharp
var config = new ConfigurationOptions
{
    FeaturesRefreshRate = 1
};

var factory = new SplitFactory("API_KEY", config);
var sdk = factory.Client();

try
{
    sdk.BlockUntilReady(10000);
}
catch (Exception ex)
{    
}

while(true)
{
    var treatment = sdk.GetTreatment("key", "feature");

    Console.WriteLine($"Value: {treatment}");

    Thread.Sleep(10000);
}
```
Repro steps

1. Start the application and wait for the 1st treatment value to show in the console
2. Change the value of the treatment
3. Wait until the new value is displayed in the console

### In version `7.4.3`
We can see that the new value is fetched right away
![image](https://github.com/splitio/dotnet-client/assets/154558268/21a9976a-a38c-4f10-9ff6-43c34e7caa8e)

### In version `7.7.0`(or any version past `7.4.3`)
The new value is never fetched no matter how long we wait.
![image](https://github.com/splitio/dotnet-client/assets/154558268/9434f112-2fc1-4a23-8027-4f3d1783be57)

**Comment by sanzmauro** [comment]
hi @michaellegare 

can you enable logs and share them with me? 🙏 

```csharp
var config = new ConfigurationOptions
{
    Logger = SplitLogger.Console(Level.Debug)
};

```

Thanks in advance,
Mauro

**Comment by michaellegare** [comment]
Hey @sanzmauro, thanks for the quick response!

Here are the logs for `7.4.3` and `7.7.0`
[split 7.7.0 logs.txt](https://github.com/splitio/dotnet-client/files/13753826/split.7.7.0.logs.txt)
[split 7.4.3 logs.txt](https://github.com/splitio/dotnet-client/files/13753827/split.7.4.3.logs.txt)

**Comment by sanzmauro** [comment]
Thanks for sharing the logs @michaellegare 

I will check what is the problem, meanwhile if you need a quick solution you can disable streaming and it should be work

```csharp
var config = new ConfigurationOptions
{
    Logger = SplitLogger.Console(Level.Debug),
    StreamingEnabled = false
};
```

**Comment by michaellegare** [comment]
Awesome!, I can confirm this workaround is working for now.

Thanks for your help @sanzmauro.

**Comment by agustinona** [comment]
Hi @michaellegare ,
In addition to @sanzmauro 's comments and workaround, from the error you are getting it seems you may be using a client side API key to initialize your SDK. If this is the case please note that the .NET SDK requires a server side API key to work properly. Using a server side API key will fix this issue and allow you to re-enable streaming in versions after 7.4.3 for faster feature flag updates.

**Comment by jonathanhallee** [comment]
@agustinona good to know thanks!. (I work with Michael). I though from the docs it didn't matter to use client side api keys in the backend since from our tests it didn't seem to be a problem

**Comment by agustinona** [comment]
@jonathanhallee Yes, BE SDKs work, to a degree, with client side API keys as well. BE SDKs require server side API keys to fetch segment information and connect to streaming. If you are not using segments in your targeting rules the SDK does not attempt to fetch segment data so it will still initialize properly (though it will fail to connect to streaming), but if you were to include segments in your targeting rules you'd start seeing issues.

**Comment by sanzmauro** [comment]
Hi @michaellegare 

I am closing the issue since we released [7.8.0](https://www.nuget.org/packages/Splitio/7.8.0) with the fix.

Regards,
Mauro
